Fix crossTestBridges

eed3si9n opened the pull request.

eed3si9n committed:
Fix crossTestBridges

typesafe-tools commented:
To validate this pull request against other sbt modules, please visit <A HREF="https://jenkins.scala-sbt.org:8592/job/sbt-validator/parambuild/?sbt=develop&zinc=pull/766/head&io=develop&librarymanagement=develop&util=develop&website=develop&deploy_to_bintray=false&run_sbt_community=false&verbose_dbuild=false&community_verbose=false">this link.</A>

eed3si9n commented:
AppVeyor failure seems legit. We need a new IO from develop branch.

eed3si9n committed:
IO 1.4.0-M3

eed3si9n commented:
```
[00:21:26] Running source-dependencies / less-inter-inv-java
[00:21:27] Running source-dependencies / constants
[00:21:27] Running apiinfo / circular-structure
[00:21:27] Running source-dependencies / sealed
[00:21:27] Running source-dependencies / value-class-underlying
[00:21:29] [debug] Reading incremental options from map
[00:21:29] [debug] [zinc] IncrementalCompile -----------
[00:21:29] [debug] Incremental.compile
[00:21:29] [debug] previous = Stamps for: 0 products, 0 sources, 0 libraries Map()
[00:21:29] [debug] current source = Set(${0}/A.scala, ${0}/D.scala)
[00:21:29] [debug] > initialChanges = InitialChanges(Changes(added = Set(${0}/A.scala, ${0}/D.scala), removed = Set(), changed = Set(), unmodified = ...),Set(),Set(),API Changes: Set())
[00:21:29] [debug] Full compilation, no sources in previous analysis.
[00:21:29] [debug] all 2 sources are invalidated
[00:21:29] [debug] Initial set of included nodes: 
[00:21:29] [info] Compiling 2 Scala sources to C:\Users\appveyor\AppData\Local\Temp\1\sbt_ba586576\target\classes ...
[00:21:29] [debug] Returning already retrieved and compiled bridge: C:\Users\appveyor\AppData\Local\Temp\1\sbt_aa3534ee\target-bridge-2.12.10.jar.
[00:21:29] [error] ## Exception when compiling 2 sources to C:\Users\appveyor\AppData\Local\Temp\1\sbt_ba586576\target\classes
[00:21:29] [error] java.lang.ClassNotFoundException: xsbt.CompilerInterface
[00:21:29] [error] java.net.URLClassLoader.findClass(URLClassLoader.java:381)
[00:21:29] [error] java.lang.ClassLoader.loadClass(ClassLoader.java:424)
[00:21:29] [error] java.lang.ClassLoader.loadClass(ClassLoader.java:357)
[00:21:29] [error] java.lang.Class.forName0(Native Method)
[00:21:29] [error] java.lang.Class.forName(Class.java:348)
[00:21:29] [error] sbt.internal.inc.AnalyzingCompiler.getInterfaceClass(AnalyzingCompiler.scala:289)
[00:21:29] [error] sbt.internal.inc.AnalyzingCompiler.call(AnalyzingCompiler.scala:256)
[00:21:29] [error] sbt.internal.inc.AnalyzingCompiler.newCachedCompiler(AnalyzingCompiler.scala:152)
[00:21:29] [error] sbt.internal.inc.AnalyzingCompiler.newCachedCompiler(AnalyzingCompiler.scala:139)
[00:21:29] [error] sbt.internal.inc.FreshCompilerCache.apply(CompilerCache.scala:103)
[00:21:29] [error] sbt.internal.inc.AnalyzingCompiler.compile(AnalyzingCompiler.scala:99)
[00:21:29] [error] sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$5(MixedAnalyzingCompiler.scala:105)
[00:21:29] [error] scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[00:21:29] [error] sbt.internal.inc.MixedAnalyzingCompiler.timed(MixedAnalyzingCompiler.scala:198)
[00:21:29] [error] sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$3(MixedAnalyzingCompiler.scala:96)
[00:21:29] [error] sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$3$adapted(MixedAnalyzingCompiler.scala:85)
[00:21:29] [error] sbt.internal.inc.JarUtils$.withPreviousJar(JarUtils.scala:226)
[00:21:29] [error] sbt.internal.inc.MixedAnalyzingCompiler.compileScala$1(MixedAnalyzingCompiler.scala:85)
[00:21:29] [error] sbt.internal.inc.MixedAnalyzingCompiler.compile(MixedAnalyzingCompiler.scala:162)
[00:21:29] [error] sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1(IncrementalCompilerImpl.scala:446)
[00:21:29] [error] sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1$adapted(IncrementalCompilerImpl.scala:446)
[00:21:29] [error] sbt.internal.inc.Incremental$.doCompile(Incremental.scala:136)
[00:21:29] [error] sbt.internal.inc.Incremental$.$anonfun$compile$6(Incremental.scala:110)
[00:21:29] [error] sbt.internal.inc.IncrementalCommon.recompileClasses(IncrementalCommon.scala:201)
[00:21:29] [error] sbt.internal.inc.IncrementalCommon.cycle(IncrementalCommon.scala:110)
[00:21:29] [error] sbt.internal.inc.Incremental$.$anonfun$compile$5(Incremental.scala:113)
[00:21:29] [error] sbt.internal.inc.Incremental$.manageClassfiles(Incremental.scala:173)
[00:21:29] [error] sbt.internal.inc.Incremental$.compile(Incremental.scala:101)
[00:21:29] [error] sbt.internal.inc.IncrementalCompile$.apply(IncrementalCompile.scala:82)
[00:21:29] [error] sbt.internal.inc.IncrementalCompilerImpl.compileInternal(IncrementalCompilerImpl.scala:452)
[00:21:29] [error] sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileIncrementally$1(IncrementalCompilerImpl.scala:401)
[00:21:29] [error] sbt.internal.inc.IncrementalCompilerImpl.handleCompilationError(IncrementalCompilerImpl.scala:264)
[00:21:29] [error] sbt.internal.inc.IncrementalCompilerImpl.compileIncrementally(IncrementalCompilerImpl.scala:347)
[00:21:29] [error] sbt.internal.inc.IncrementalCompilerImpl.compile(IncrementalCompilerImpl.scala:79)
[00:21:29] [error] sbt.internal.inc.ProjectStructure.compile(IncHandler.scala:507)
[00:21:29] [error] sbt.internal.inc.IncHandler$$anonfun$commands$lzycompute$1.applyOrElse(IncHandler.scala:181)
[00:21:29] [error] sbt.internal.inc.IncHandler$$anonfun$commands$lzycompute$1.applyOrElse(IncHandler.scala:181)
[00:21:29] [error] sbt.internal.inc.IncHandler.applyOrElse(IncHandler.scala:254)
[00:21:29] [error] sbt.internal.inc.IncHandler.$anonfun$noArgs$1(IncHandler.scala:252)
[00:21:29] [error] sbt.internal.inc.IncHandler.$anonfun$noArgs$1$adapted(IncHandler.scala:252)
[00:21:29] [error] sbt.internal.inc.IncHandler.$anonfun$apply$1(IncHandler.scala:144)
[00:21:29] [error] sbt.internal.inc.IncHandler.$anonfun$apply$1$adapted(IncHandler.scala:144)
[00:21:29] [error] sbt.internal.inc.IncHandler.onIncInstance(IncHandler.scala:152)
[00:21:29] [error] sbt.internal.inc.IncHandler.apply(IncHandler.scala:145)
[00:21:29] [error] sbt.internal.inc.IncHandler.apply(IncHandler.scala:63)
[00:21:29] [error] sbt.internal.inc.BatchScriptRunner.processStatement(BatchScriptRunner.scala:45)
[00:21:29] [error] sbt.internal.inc.BatchScriptRunner.$anonfun$apply$1(BatchScriptRunner.scala:27)
[00:21:29] [error] sbt.internal.inc.BatchScriptRunner.$anonfun$apply$1$adapted(BatchScriptRunner.scala:27)
[00:21:29] [error] scala.collection.immutable.List.foreach(List.scala:392)
[00:21:29] [error] sbt.internal.inc.BatchScriptRunner.apply(BatchScriptRunner.scala:27)
[00:21:29] [error] sbt.internal.inc.ScriptedTests.$anonfun$commonRunTest$8(ScriptedTests.scala:234)
[00:21:29] [error] scala.util.control.Exception$Catch.apply(Exception.scala:228)
[00:21:29] [error] sbt.internal.inc.ScriptedTests.commonRunTest(ScriptedTests.scala:231)
[00:21:29] [error] sbt.internal.inc.ScriptedTests.$anonfun$runBatchedTests$2(ScriptedTests.scala:163)
[00:21:29] [error] sbt.internal.inc.ScriptedTests.runOrHandleDisabled(ScriptedTests.scala:181)
[00:21:29] [error] sbt.internal.inc.ScriptedTests.$anonfun$runBatchedTests$1(ScriptedTests.scala:164)
[00:21:29] [error] scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:238)
[00:21:29] [error] scala.collection.mutable.ArraySeq.foreach(ArraySeq.scala:75)
[00:21:29] [error] scala.collection.TraversableLike.map(TraversableLike.scala:238)
[00:21:29] [error] scala.collection.TraversableLike.map$(TraversableLike.scala:231)
[00:21:29] [error] scala.collection.AbstractTraversable.map(Traversable.scala:108)
[00:21:29] [error] sbt.internal.inc.ScriptedTests.runBatchTests$1(ScriptedTests.scala:147)
[00:21:29] [error] sbt.internal.inc.ScriptedTests.runBatchedTests(ScriptedTests.scala:170)
[00:21:29] [error] sbt.internal.inc.ScriptedTests.$anonfun$batchScriptedRunner$10(ScriptedTests.scala:82)
[00:21:29] [error] sbt.io.IO$.withTemporaryDirectory(IO.scala:490)
[00:21:29] [error] sbt.io.IO$.withTemporaryDirectory(IO.scala:500)
[00:21:29] [error] sbt.internal.inc.ScriptedTests.$anonfun$batchScriptedRunner$9(ScriptedTests.scala:81)
[00:21:29] [error] sbt.internal.inc.ScriptedRunnerImpl$.$anonfun$runAllInParallel$1(IncScriptedRunner.scala:66)
[00:21:29] [error] scala.collection.parallel.mutable.ParArray$ParArrayIterator.flatmap2combiner(ParArray.scala:419)
[00:21:29] [error] scala.collection.parallel.ParIterableLike$FlatMap.leaf(ParIterableLike.scala:1074)
[00:21:29] [error] scala.collection.parallel.Task.$anonfun$tryLeaf$1(Tasks.scala:53)
[00:21:29] [error] scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[00:21:29] [error] scala.util.control.Breaks$$anon$1.catchBreak(Breaks.scala:67)
[00:21:29] [error] scala.collection.parallel.Task.tryLeaf(Tasks.scala:56)
[00:21:29] [error] scala.collection.parallel.Task.tryLeaf$(Tasks.scala:50)
[00:21:29] [error] scala.collection.parallel.ParIterableLike$FlatMap.tryLeaf(ParIterableLike.scala:1070)
[00:21:29] [error] scala.collection.parallel.AdaptiveWorkStealingTasks$WrappedTask.internal(Tasks.scala:160)
[00:21:29] [error] scala.collection.parallel.AdaptiveWorkStealingTasks$WrappedTask.internal$(Tasks.scala:157)
[00:21:29] [error] scala.collection.parallel.AdaptiveWorkStealingForkJoinTasks$WrappedTask.internal(Tasks.scala:440)
[00:21:29] [error] scala.collection.parallel.AdaptiveWorkStealingTasks$WrappedTask.compute(Tasks.scala:150)
[00:21:29] [error] scala.collection.parallel.AdaptiveWorkStealingTasks$WrappedTask.compute$(Tasks.scala:149)
[00:21:29] [error] scala.collection.parallel.AdaptiveWorkStealingForkJoinTasks$WrappedTask.compute(Tasks.scala:440)
[00:21:29] [error] java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:189)
[00:21:29] [error] java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
[00:21:29] [error] java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
[00:21:29] [error] java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
[00:21:29] [error] java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
```
